core/src/main/scala/cats/data/Op.scala

```diff
@@ -23,11 +23,60 @@ package cats
 package data
 
 import cats.arrow.*
+import cats.kernel.Eq
+import cats.arrow.Compose
```

Comment:
Let's rollback this change. It is neither required for the PR nor improves anything in the file.

Otherwise the PR looks pretty good, thank you for your dedication!